packages/wallets/src/components/WalletTransactionsCompleted/WalletTransactionsCompleted.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useCallback, useEffect } from 'react';
+import moment from 'moment';
+import { useAuthorize, useInvalidateQuery, useTransactions } from '@deriv/api';
+import { TSocketRequestPayload } from '@deriv/api/types';
+import { WalletTransactionsCompletedRow } from '../WalletTransactionsCompletedRow';
+import { WalletTransactionsNoDataState } from '../WalletTransactionsNoDataState';
+import { WalletTransactionsTable } from '../WalletTransactionsTable';
+
+type TFilter = NonNullable<TSocketRequestPayload<'statement'>['payload']>['action_type'];
+
+type TProps = {
+    filter?: TFilter;
+    setFilter: (value?: TFilter) => void;
+};
+
+const filtersMapper: Record<string, Partial<TProps['filter']>> = {
+    all: undefined,
+    deposit: 'deposit',
+    transfer: 'transfer',
+    withdrawal: 'withdrawal',
+};
+
+const WalletTransactionsCompleted: React.FC<TProps> = ({ filter, setFilter }) => {
+    const { isFetching: isAuthorizeFetching } = useAuthorize();
+    const invalidate = useInvalidateQuery();
+    const { data, fetchNextPage, isFetching, setFilter: setTransactionsFilter } = useTransactions();
+
+    const fetchMoreOnBottomReached = useCallback(
+        (containerRefElement?: HTMLDivElement | null) => {
+            if (containerRefElement) {
+                const { clientHeight, scrollHeight, scrollTop } = containerRefElement;
+                //once the user has scrolled within 300px of the bottom of the table, fetch more data if there is any
+                if (scrollHeight - scrollTop - clientHeight < 300 && !isFetching) {
+                    fetchNextPage();
+                }
+            }
+        },
+        [fetchNextPage, isFetching]
+    );
+
+    useEffect(() => {
+        setFilter(filtersMapper[filter || 'all']);
```

Comment:
which setFilter is being called here? the one passed as prop or the one returned by "useTransactions" hook?

Reply:
in my mind, ideally, we shouldn't need the prop "setFilter" in WalletTransactionsCompleted (and in WalletTransactionsPending), as this state is owned by WalletTransactions component